waspc/cli/Command/Compile.hs

```diff
@@ -0,0 +1,35 @@
+module Command.Compile
+    ( compile
+    ) where
+
+import           Control.Monad.IO.Class (liftIO)
+import           Data.List              (find, isSuffixOf)
+import qualified Path                   as P
+
+import qualified Common
+import           CompileOptions         (CompileOptions (..))
+import qualified Lib
+import           StrongPath             (Abs, Dir, Path, (</>))
+import qualified StrongPath             as SP
+import qualified Util.IO
+
+
+-- | Compiles Wasp source code in waspRoot directory and generates a project in given outDir directory.
+compile :: Path Abs (Dir Common.WaspProjectDir) -> Path Abs (Dir Lib.ProjectRootDir) -> IO (Either String ())
+compile waspRoot outDir = do
+    maybeWaspFile <- findWaspFile waspRoot
+    case maybeWaspFile of
+        Nothing -> return $ Left "No .wasp file present in the root of Wasp project."
+        Just waspFile -> Lib.compile waspFile outDir options
+  where
+    options = CompileOptions
+        { externalCodeDirPath = waspRoot </> Common.extCodeDirInWaspProjectDir }
+
+    findWaspFile :: Path Abs (Dir d) -> IO (Maybe (Path Abs SP.File))
+    findWaspFile dir = do
+        (files, _) <- liftIO $ Util.IO.listDirectory (SP.toPathAbsDir dir)
+        return $ (dir SP.</>) . SP.fromPathRelFile <$> find isWaspFile files
+
+    isWaspFile :: P.Path P.Rel P.File -> Bool
```

Comment:
didn't we already before somewhere have this function? I remember reviewing it already.

Reply:
Yes I just moved it now :D.